crates/swc_ecma_minifier/tests/fixture/issues/7847/output.js

```diff
@@ -0,0 +1,9 @@
+if (g()) {
+    var state, hasRequiredState;
+    const a = hasRequiredState ? state : (hasRequiredState = 1, state = {
+        getHighWaterMark: function(o) {
+            return o.objectMode ? 16 : 16384;
+        }
+    });
+    console.log(a.getHighWaterMark());
```

Comment:
Nit: Can we add a `console.log(state)` here? The old transform is technically correct, since `state` is never read after the assignment (this could have been eliminated with a lexical analysis)

Reply:
@jridgewell I prefer this:

```js
// generated by @rollup/plugin-commonjs
function requireState() {
    if (hasRequiredState) return state;
    hasRequiredState = 1;
    return state = {
        getHighWaterMark: function(o) {
            return o.objectMode ? 16 : 16384;
        }
    };
}
if (g()) {
    var state, hasRequiredState;
    const a = requireState();
	console.log(a.getHighWaterMark());
    // Hidden state reading!
    const b = requireState();
	console.log(b.getHighWaterMark());
}
```

This is from my original issue, where swc incorrectly transform this into:

```js
'use strict';
function requireState() {
    return hasRequiredState ? state : (hasRequiredState = 1, {
        getHighWaterMark: function(o) {
            return o.objectMode ? 16 : 16384;
        }
    });
}
if (g()) {
    var state, hasRequiredState;
    let a = requireState();
    console.log(a.getHighWaterMark());
    let b = requireState();
    console.log(b.getHighWaterMark());
}
```

Reply:
Done